File: ZooJava/src/main/java/com/example/zoojava/utils/csv/ImportAnimalCsvImpl.java
package com.example.zoojava.utils.csv;

import com.example.zoojava.models.Animal;
import com.example.zoojava.models.enums.typeAnimal;

import javax.inject.Inject;
import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.time.LocalDate;
import java.util.List;
import java.util.stream.Collectors;

public class ImportAnimalCsvImpl implements ImportAnimalCsv {
    private final Path CURRENT = Paths.get("");
    private final String DIR = CURRENT.toAbsolutePath() + File.separator + "csv";
    private final String FILE_CSV = DIR + File.separator + "animales.csv";


    @Override
    public List<Animal> importData() {
        List<Animal> lista;
        try {
            lista= Files.lines(Path.of(FILE_CSV)).skip(1).map(this::changeToAnimal).collect(Collectors.toList());
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
        return lista;
    }

    private Animal changeToAnimal(String line){
        var campos = line.split(";");
        var id = Integer.parseInt(campos[0]);
        var name = campos[1];
        var type = typeAnimal.valueOf(campos[2]);
        var birthDate = parseFecha(campos[3]);
        var img = campos[4];
        return new Animal(id, name, type, birthDate,img);
    }


    private LocalDate parseFecha(String campo) {
        var fecha = campo.split("/");
        var dia = Integer.parseInt(fecha[0]);
        var mes = Integer.parseInt(fecha[1]);
        var year = Integer.parseInt(fecha[2]);
        return LocalDate.of(year,mes,dia);
    }
}

File: ZooKotlin/src/main/kotlin/com/example/zookotlin/utils/csv/ImportAnimalCsvImpl.kt
package com.example.zookotlin.utils.csv

import com.example.zookotlin.models.Animal
import com.example.zookotlin.models.enums.typeAnimal
import com.example.zookotlin.utils.Globals
import java.nio.file.Files
import java.nio.file.Path
import java.time.LocalDate
import java.util.stream.Collectors

class ImportAnimalCsvImpl: ImportAnimalCsv {
    override fun importData(): List<Animal> {
        var list = Files.lines(Path.of(Globals.ANIMALS_CSV)).skip(1).map { line: String ->
            this.changeToAnimal(
                line
            )
        }.collect(Collectors.toList())
        return list
    }


    private fun changeToAnimal(line: String): Animal {
        val campos = line.split(";")
        val id = campos[0].toInt()
        val name = campos[1]
        val type = typeAnimal.valueOf(campos[2])
        val birthDate = parseFecha(campos[3])
        val img = campos[4]
        return Animal(id, name, type, birthDate, img)
    }


    private fun parseFecha(campo: String): LocalDate {
        val fecha = campo.split("-")
        val dia = fecha[2].toInt()
        val mes = fecha[1].toInt()
        val year = fecha[0].toInt()
        return LocalDate.of(year, mes, dia)
    }
}